internal/store/access_role.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/config"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+//go:generate mockgen -destination=../mocks/mock_access_role.go -package=mocks github.com/mongodb/mongocli/internal/store AccessRoleCreator
+
+type AccessRoleCreator interface {
+	AccessRole(string, string) (*atlas.AWSIAMRole, error)
+}
```

Comment:
[nit] we need to bit a bit more specific here
```suggestion
type CloudProviderAccessRoleCreator interface {
	CreateCloudProviderAccessRole(string, string) (*atlas.AWSIAMRole, error)
}
```